profile.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package eat
+
+import (
+	"encoding/asn1"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// Profile is either a well formed RFC3986 compliant URI(uri  *url.URL).
+// or an ObjectIdentifier  ([]int)
+type Profile struct {
+	val interface{}
+}
+
+// NewProfile instantiates a new Profile object from a given input string
+// The input string can either be a well formed URI as string
+// OR it could be a OID formated as a string example "1.2.3.4"
+func NewProfile(urlOrOID string) (*Profile, error) {
+	p := Profile{}
+	if err := p.SetProfile(urlOrOID); err != nil {
+		return nil, err
+	}
+	return &p, nil
+}
+
+// SetProfile sets the given urlOrOID overwriting a previously stored
+// value
+func (s *Profile) SetProfile(urlOrOID string) error {
+	// First attempt to decode input string as a URL
+	u, err := url.Parse(urlOrOID)
+	if err != nil || !u.IsAbs() {
+		s.val, err = decodeOIDfromString(urlOrOID)
+		if err != nil {
+			return fmt.Errorf("no valid URI or OID supplied as an argument: %w", err)
+		}
+	} else {
+		s.val = u
+	}
+	return nil
+}
+
+// GetProfile returns either a valid URI or OID as strings
+// to the caller
+func (s Profile) GetProfile() (string, error) {
+	switch t := s.val.(type) {
+	case *url.URL:
+		return t.String(), nil
+	case asn1.ObjectIdentifier:
+		return t.String(), nil
+	default:
+		return "", fmt.Errorf("no valid EAT profile")
+	}
+}
+
+// IsURI checks whether a stored profile is a URI
+func (s Profile) IsURI() bool {
+	switch s.val.(type) {
+	case *url.URL:
+		return true
+	default:
+		return false
+	}
+}
+
+// IsOID checks whether a stored profile is an OID
+func (s Profile) IsOID() bool {
+	switch s.val.(type) {
+	case asn1.ObjectIdentifier:
+		return true
+	default:
+		return false
+	}
+}
+
+// DecodeProfileCBOR decodes from a received CBOR data the profile
+// as either a URL or a Object Identifier
+func (s *Profile) DecodeProfileCBOR(val interface{}) error {
```

Comment:
this is definitely not a public method
```suggestion
func (s *Profile) decodeProfileCBOR(val interface{}) error {
```